File: playwright-dotnet/tests/playwright-ts/pages/loginPage.ts
import { Page } from '@playwright/test';
import { BasePage } from './basePage';

/**
 * LoginPage — wraps SauceDemo login form interactions.
 * Selectors mirror C# LoginPage.cs.
 */
export class LoginPage extends BasePage {
  private readonly usernameInput = '#user-name';
  private readonly passwordInput = '#password';
  private readonly loginButton   = '#login-button';
  private readonly errorMessage  = "h3[data-test='error']";

  constructor(page: Page) {
    super(page);
  }

  async loginAs(username: string, password: string): Promise<void> {
    await this.fill(this.usernameInput, username);
    await this.fill(this.passwordInput, password);
    await this.click(this.loginButton);
  }

  async getErrorMessage(): Promise<string> {
    return await this.getText(this.errorMessage);
  }

  async isErrorDisplayed(): Promise<boolean> {
    return await this.isVisible(this.errorMessage);
  }

  async isLoginButtonVisible(): Promise<boolean> {
    return await this.isVisible(this.loginButton);
  }
}

File: cucumber/src/main/java/com/cucumber/framework/pages/LoginPage.java
package com.cucumber.framework.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * LoginPage — SauceDemo authentication.
 * Selectors are identical across all three frameworks in this monorepo.
 */
public class LoginPage extends BasePage {

    private final By usernameInput = By.id("user-name");
    private final By passwordInput = By.id("password");
    private final By loginButton   = By.id("login-button");
    private final By errorMessage  = By.cssSelector("h3[data-test='error']");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void login(String username, String password) {
        fill(usernameInput, username);
        fill(passwordInput, password);
        click(loginButton);
    }

    public String getErrorMessage() {
        return getText(errorMessage);
    }

    public boolean isErrorDisplayed() {
        return isVisible(errorMessage);
    }
}
